user-manager/src/test/java/com/qpay/usermanager/repository/CustomerRepositoryTest.java

```diff
@@ -0,0 +1,133 @@
+package com.qpay.usermanager.repository;
+
+import com.qpay.usermanager.model.entity.customer.CustomerEntity;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+class CustomerRepositoryTest {
+
+    @Autowired
+    private CustomerRepository customerRepository;
+
+    @Autowired
+    private TestEntityManager testEntityManager;
+
+    @Test
+    void should_returnCustomerByEmail() {
+        // given
+        var email = "admin@gmail.com";
+        var customer1 = testEntityManager.persistAndFlush(
+                CustomerEntity.builder()
+                        .name("Roman")
+                        .email("admin@gmail.com")
+                        .password("password")
+                        .walletId(1L)
+                        .build()
+        );
+        var customer2 = testEntityManager.persistAndFlush(
+                CustomerEntity.builder()
+                        .name("Anatolyi")
+                        .email("user@gmail.com")
+                        .password("password")
+                        .walletId(2L)
+                        .build()
+        );
+
+        // when
+        var result = customerRepository.findCustomerEntityByEmail(email);
+
+        // then
+        assertThat(result).isEqualTo(customer1);
+    }
+
+    @Test
+    void should_returnNull_IfCustomerDoesNotExistByEmail() {
```

Comment:
`should_returnNull_when_CustomerDoesNotExisWithGivenEmail()`
replace `if` with `when` in all other cases